lib/realtime_web/channels/realtime_channel/logging.ex

```diff
@@ -9,45 +9,28 @@ defmodule RealtimeWeb.RealtimeChannel.Logging do
   @doc """
   Checks if the log level set in the socket is less than or equal to the given level of the message to be logged.
   """
-  @spec maybe_log(
-          socket :: Phoenix.Socket.t(),
-          level :: Logger.level(),
-          code :: binary(),
-          msg :: binary(),
-          metadata :: keyword()
-        ) :: :ok
-  def maybe_log(%{assigns: %{log_level: log_level}}, level, code, msg, metadata \\ []) do
-    metadata = if metadata == [], do: Logger.metadata()
-
+  @spec maybe_log(socket :: Phoenix.Socket.t(), level :: Logger.level(), code :: binary(), msg :: binary()) :: :ok
+  def maybe_log(%{assigns: %{log_level: log_level, tenant: tenant}}, level, code, msg) do
+    Logger.metadata(external_id: tenant, project: tenant)
     msg = stringify!(msg)
-
-    if Logger.compare_levels(log_level, level) != :gt do
-      Logger.log(level, "#{code}: #{msg}", metadata)
-    end
+    if Logger.compare_levels(log_level, level) != :gt, do: Logger.log(level, "#{code}: #{msg}")
   end
 
-  def maybe_log_error(socket, code, msg, metadata \\ []), do: maybe_log(socket, :error, code, msg, metadata)
-  def maybe_log_warning(socket, code, msg, metadata \\ []), do: maybe_log(socket, :warning, code, msg, metadata)
+  def maybe_log_error(socket, code, msg), do: maybe_log(socket, :error, code, msg)
+  def maybe_log_warning(socket, code, msg), do: maybe_log(socket, :warning, code, msg)
 
-  def maybe_log_info(%{assigns: %{log_level: log_level}}, msg, metadata \\ []) do
-    metadata = if metadata == [], do: Logger.metadata()
-
-    if Logger.compare_levels(log_level, :info) != :gt do
-      Logger.info(inspect(msg), metadata)
-    end
+  def maybe_log_info(%{assigns: %{log_level: log_level, tenant: tenant}}, msg) do
+    Logger.metadata(external_id: tenant, project: tenant)
```

Comment:
Same as the other PR. I think this should be set at when the process start like we do on `RealtimeChannel.join`. This will keep trying to update the process metadata which will just be a waste operation I think. We probably don't need to touch the metadata ?